lib/orderbook_sync.js

```diff
@@ -1,25 +1,36 @@
 const WebsocketClient = require('./clients/websocket.js');
+const AuthenticatedClient = require('./clients/authenticated.js');
 const PublicClient = require('./clients/public.js');
 const Orderbook = require('./orderbook.js');
+const Utils = require('./utilities.js');
 
 // Orderbook syncing
 class OrderbookSync extends WebsocketClient {
   constructor(
     productIDs,
     apiURI = 'https://api.gdax.com',
     websocketURI = 'wss://ws-feed.gdax.com',
-    authenticatedClient = null,
+    auth = null,
     { heartbeat = false } = {}
   ) {
-    super(productIDs, websocketURI, authenticatedClient, { heartbeat });
+    super(productIDs, websocketURI, auth, { heartbeat });
     this.apiURI = apiURI;
-    this.authenticatedClient = authenticatedClient;
+    this.auth = Utils.checkAuth(auth);
 
     this._queues = {}; // []
     this._sequences = {}; // -1
     this._public_clients = {};
     this.books = {};
 
+    if (this.auth.secret) {
+      this._authenticatedClient = new AuthenticatedClient(
```

Comment:
Can this default to `auth` if it already is an instance of `AuthenticatedClient`?

Reply:
I vote note. For the same reasons as above. We should not be making side-effect assumptions about the passed object, aside from the fact that it quacks like a `{ key, secret, passphrase }` object. Otherwise, the project opens itself to a world of forwards compatibility issues and limitations down the road. 

i.e. "Optimizations" like this only yield future maintenance problems. There's almost zero overhead to having an extra `AuthenticatedClient` instance. 

e.g. If we did reuse the passed-in `AuthenticationClient`, the `apiURI` argument passed to the `OrderbookSync` constructor would then be ignored, and you'd end up with unintended consequences of talking to the wrong endpoint.